### PR TITLE
[Feature] Fix the validation of function privilege entry object

### PR DESCRIPTION
**Pull request by nshangyiming**
## What type of PR is this：
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

This is to fix the bug of previous pr #14475

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/privilege/PrivilegeCollection.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/privilege/FunctionPEntryObject.java|1|1|100.00%|[]|

**Review by Astralidea — APPROVED** [review]

File: fe/fe-core/src/main/java/com/starrocks/privilege/PrivilegeCollection.java [hunk 1]
```diff
@@ -272,7 +272,7 @@ public void removeInvalidObject(GlobalStateMgr globalStateMgr) {
                 PrivilegeEntry entry = entryIterator.next();
                 if (entry.object != null && !entry.object.isFuzzyMatching() && !entry.object.validate(globalStateMgr)) {
                     String entryStr = GsonUtils.GSON.toJson(entry);
-                    LOG.info("find invalidate object, will remove the entry now: {}", entryStr);
+                    LOG.info("find invalid object, will remove the entry now: {}", entryStr);
```
Comment (RowenWood) [inline fe/fe-core/src/main/java/com/starrocks/privilege/PrivilegeCollection.java:275]: why info level log？？

**Review by gengjun-git — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

File: fe/fe-core/src/main/java/com/starrocks/privilege/PrivilegeCollection.java (hunk 1 above)
Comment (nshangyiming) [reply fe/fe-core/src/main/java/com/starrocks/privilege/PrivilegeCollection.java:275]: Considering the situation we are removing a privilege entry object, it's necessary.

Comment (nshangyiming) [reply fe/fe-core/src/main/java/com/starrocks/privilege/PrivilegeCollection.java:275]: It's not a common log.

**Pull request merged**